diff --git a/drivers/net/wireless/rockchip_wlan/Kconfig b/drivers/net/wireless/rockchip_wlan/Kconfig
index 858f0d6e6..9ce4d8ea0 100644
--- a/drivers/net/wireless/rockchip_wlan/Kconfig
+++ b/drivers/net/wireless/rockchip_wlan/Kconfig
@@ -41,5 +41,7 @@ source "drivers/net/wireless/rockchip_wlan/infineon/Kconfig"
 
 source "drivers/net/wireless/rockchip_wlan/8897U/wlan_src/Kconfig"
 source "drivers/net/wireless/rockchip_wlan/8897U/mbtc_src/Kconfig"
+source "drivers/net/wireless/rockchip_wlan/rtl8821cs/Kconfig"
+
 
 endif # WL_ROCKCHIP
diff --git a/drivers/net/wireless/rockchip_wlan/Makefile b/drivers/net/wireless/rockchip_wlan/Makefile
index 6439eba8b..428550c9c 100644
--- a/drivers/net/wireless/rockchip_wlan/Makefile
+++ b/drivers/net/wireless/rockchip_wlan/Makefile
@@ -4,3 +4,4 @@ obj-$(CONFIG_CYW_BCMDHD)	+= cywdhd/
 obj-$(CONFIG_INFINEON_DHD)	+= infineon/
 obj-$(CONFIG_MRVL8897U)        += 8897U/wlan_src/
 obj-$(CONFIG_MRVL8897U_BT)      += 8897U/mbtc_src/
+obj-$(CONFIG_RTL8821CS) += rtl8821cs/
diff --git a/drivers/net/wireless/rockchip_wlan/rtl8821cs/Makefile b/drivers/net/wireless/rockchip_wlan/rtl8821cs/Makefile
index 797d011e6..bbfeab2d8 100644
--- a/drivers/net/wireless/rockchip_wlan/rtl8821cs/Makefile
+++ b/drivers/net/wireless/rockchip_wlan/rtl8821cs/Makefile
@@ -11,11 +11,10 @@ EXTRA_CFLAGS += -O1
 EXTRA_CFLAGS += -Wno-unused-variable
 #EXTRA_CFLAGS += -Wno-unused-value
 #EXTRA_CFLAGS += -Wno-unused-label
-#EXTRA_CFLAGS += -Wno-unused-parameter
+EXTRA_CFLAGS += -Wno-unused-parameter
 #EXTRA_CFLAGS += -Wno-unused-function
 #EXTRA_CFLAGS += -Wno-unused
 #EXTRA_CFLAGS += -Wno-uninitialized
-
 GCC_VER_49 := $(shell echo `$(CC) -dumpversion | cut -f1-2 -d.` \>= 4.9 | bc )
 ifeq ($(GCC_VER_49),1)
 EXTRA_CFLAGS += -Wno-date-time	# Fix compile error && warning on gcc 4.9 and later
@@ -23,8 +22,17 @@ endif
 
 EXTRA_CFLAGS += -I$(src)/include
 
+EXTRA_CFLAGS += -I$(src)/hal/phydm
+
 EXTRA_LDFLAGS += --strip-debug
 
+BCMDHD_ROOT = $(src)
+#$(warning "BCMDHD_ROOT=$(BCMDHD_ROOT)")
+#EXTRA_CFLAGS = $(DHDCFLAGS)
+EXTRA_CFLAGS += -DDHD_COMPILED=\"$(BCMDHD_ROOT)\"
+EXTRA_CFLAGS += -I$(srctree)/$(BCMDHD_ROOT)/include/ -I$(srctree)/$(BCMDHD_ROOT)/ -I$(srctree)/$(BCMDHD_ROOT)/hal/phydm/
+EXTRA_CFLAGS += $(call cc-disable-warning, unused-but-set-variable)
+
 CONFIG_AUTOCFG_CP = n
 
 ########################## WIFI IC ############################
@@ -94,7 +102,7 @@ CONFIG_RTW_MBO = n
 
 ########################## Android ###########################
 # CONFIG_RTW_ANDROID - 0: no Android, 4/5/6/7/8/9/10/11 : Android version
-CONFIG_RTW_ANDROID = 11
+CONFIG_RTW_ANDROID = 0
 
 ifeq ($(shell test $(CONFIG_RTW_ANDROID) -gt 0; echo $$?), 0)
 EXTRA_CFLAGS += -DCONFIG_RTW_ANDROID=$(CONFIG_RTW_ANDROID)
@@ -104,7 +112,7 @@ endif
 CONFIG_RTW_DEBUG = y
 # default log level is _DRV_INFO_ = 4,
 # please refer to "How_to_set_driver_debug_log_level.doc" to set the available level.
-CONFIG_RTW_LOG_LEVEL = 2
+CONFIG_RTW_LOG_LEVEL = 0
 
 # enable /proc/net/rtlxxxx/ debug interfaces
 CONFIG_PROC_DEBUG = y
@@ -271,10 +279,15 @@ _HAL_INTFS_FILES :=	hal/hal_intf.o \
 
 
 EXTRA_CFLAGS += -I$(src)/platform
+
+EXTRA_CFLAGS += -I$(srctree)/$(BCMDHD_ROOT)/platform
+
 _PLATFORM_FILES := platform/platform_ops.o
 
 EXTRA_CFLAGS += -I$(src)/hal/btc
 
+EXTRA_CFLAGS += -I$(srctree)/$(BCMDHD_ROOT)/hal/btc
+
 ########### HAL_RTL8188E #################################
 ifeq ($(CONFIG_RTL8188E), y)
 
@@ -1709,9 +1722,10 @@ ifeq ($(CONFIG_SDIO_HCI), y)
 EXTRA_CFLAGS += -DRTW_SUPPORT_PLATFORM_SHUTDOWN
 endif
 # default setting for Special function
-ARCH := arm
-CROSS_COMPILE := /home/android_sdk/Rockchip/Rk3188/prebuilts/gcc/linux-x86/arm/arm-eabi-4.6/bin/arm-eabi-
-KSRC := /home/android_sdk/Rockchip/Rk3188/kernel
+#ARCH := arm
+#CROSS_COMPILE := /home/android_sdk/Rockchip/Rk3188/prebuilts/gcc/linux-x86/arm/arm-eabi-4.6/bin/arm-eabi-
+#KSRC := /home/android_sdk/Rockchip/Rk3188/kernel
+
 MODULE_NAME := 8821cs
 endif
 
@@ -2121,6 +2135,7 @@ EXTRA_CFLAGS += -DCONFIG_RADIO_WORK
 ifeq ($(CONFIG_RTL8821C)$(CONFIG_SDIO_HCI),yy)
 EXTRA_CFLAGS += -DCONFIG_WAKEUP_GPIO_INPUT_MODE
 EXTRA_CFLAGS += -DCONFIG_BT_WAKE_HST_OPEN_DRAIN
+EXTRA_CFLAGS += -DCONFIG_LITTLE_ENDIAN
 endif
 EXTRA_CFLAGS += -Wno-error=date-time
 # default setting for Android 7.0
@@ -2279,7 +2294,6 @@ endif
 
 ARCH ?= arm64
 CROSS_COMPILE ?=
-KSRC ?=
 
 ifeq ($(CONFIG_RTL8822B), y)
 ifeq ($(CONFIG_SDIO_HCI), y)
@@ -2329,6 +2343,10 @@ ifneq ($(USER_MODULE_NAME),)
 MODULE_NAME := $(USER_MODULE_NAME)
 endif
 
+ARCH = arm64
+#CROSS_COMPILE =/home/fq/code/rk356x_yocto/meta-rockchip/external/prebuilts/gcc/linux-x86/aarch64/gcc-arm-10.3-2021.07-x86_64-aarch64-none-linux-gnu/bin/aarch64-none-linux-gnu-
+#KSRC =/home/fq/code/rk356x_yocto/meta-rockchip/external/kernel
+
 ifneq ($(KERNELRELEASE),)
 
 ########### this part for *.mk ############################
@@ -2407,6 +2425,8 @@ rtk_core += core/rtw_sdio.o
 endif
 
 EXTRA_CFLAGS += -I$(src)/core/crypto
+
+EXTRA_CFLAGS += -I$(srctree)/$(BCMDHD_ROOT)/core/crypto
 rtk_core += \
 		core/crypto/aes-internal.o \
 		core/crypto/aes-internal-enc.o \
@@ -2440,14 +2460,22 @@ ifeq ($(CONFIG_RTL8723B), y)
 $(MODULE_NAME)-$(CONFIG_MP_INCLUDED)+= core/rtw_bt_mp.o
 endif
 
-obj-$(CONFIG_RTL8821CS) := $(MODULE_NAME).o
-
-else
+ifeq ($(CONFIG_RTL8821C),y)
+	EXTRA_LDFLAGS += --strip-debug
+endif
 
-export CONFIG_RTL8821CS = m
+#BCMDHD_ROOT = $(src)
+#$(warning "BCMDHD_ROOT=$(BCMDHD_ROOT)")
+#EXTRA_CFLAGS = $(DHDCFLAGS)
+EXTRA_CFLAGS += -DDHD_COMPILED=\"$(BCMDHD_ROOT)\"
+EXTRA_CFLAGS += -I$(srctree)/$(BCMDHD_ROOT)/include/ -I$(srctree)/$(BCMDHD_ROOT)/
+EXTRA_CFLAGS += $(call cc-disable-warning, unused-but-set-variable)
 
-all: modules
+obj-$(CONFIG_RTL8821CS) += $(MODULE_NAME).o
+#obj-m := $(MODULE_NAME).o
 
+ccflags-y := $(EXTRA_CFLAGS)
+else
 modules:
 	$(MAKE) ARCH=$(ARCH) CROSS_COMPILE=$(CROSS_COMPILE) -C $(KSRC) M=$(shell pwd)  modules
 
diff --git a/drivers/net/wireless/rockchip_wlan/rtl8821cs/core/efuse/rtw_efuse.c b/drivers/net/wireless/rockchip_wlan/rtl8821cs/core/efuse/rtw_efuse.c
index 1f301e651..09df60092 100644
--- a/drivers/net/wireless/rockchip_wlan/rtl8821cs/core/efuse/rtw_efuse.c
+++ b/drivers/net/wireless/rockchip_wlan/rtl8821cs/core/efuse/rtw_efuse.c
@@ -938,16 +938,6 @@ void rtw_efuse_analyze(PADAPTER	padapter, u8 Type, u8 Fake)
 	i = 0;
 	j = 0;
 
-	for (i = 0; i < mapLen; i++) {
-		if (i % 16 == 0)
-			RTW_PRINT_SEL(RTW_DBGDUMP, "0x%03x: ", i);
-			_RTW_PRINT_SEL(RTW_DBGDUMP, "%02X%s"
-				, pEfuseHal->fakeEfuseInitMap[i]
-				, ((i + 1) % 16 == 0) ? "\n" : (((i + 1) % 8 == 0) ? "	  " : " ")
-			);
-		}
-	_RTW_PRINT_SEL(RTW_DBGDUMP, "\n");
-
 out_free_buffer:
 	if (eFuseWord)
 		rtw_mfree((u8 *)eFuseWord, EFUSE_MAX_SECTION_NUM * (EFUSE_MAX_WORD_UNIT * 2));
diff --git a/drivers/net/wireless/rockchip_wlan/rtl8821cs/core/rtw_pwrctrl.c b/drivers/net/wireless/rockchip_wlan/rtl8821cs/core/rtw_pwrctrl.c
index cc3101867..4da4d1ca1 100644
--- a/drivers/net/wireless/rockchip_wlan/rtl8821cs/core/rtw_pwrctrl.c
+++ b/drivers/net/wireless/rockchip_wlan/rtl8821cs/core/rtw_pwrctrl.c
@@ -90,7 +90,7 @@ void _ips_enter(_adapter *padapter)
 
 	if (rf_off == pwrpriv->change_rfpwrstate) {
 		pwrpriv->bpower_saving = _TRUE;
-		RTW_PRINT("nolinked power save enter\n");
+		//RTW_PRINT("nolinked power save enter\n");
 
 		if (pwrpriv->ips_mode == IPS_LEVEL_2)
 			pwrpriv->bkeepfwalive = _TRUE;
@@ -143,7 +143,7 @@ int _ips_leave(_adapter *padapter)
 		pwrpriv->pwr_saving_time += rtw_get_passing_time_ms(pwrpriv->pwr_saving_start_time);
 #endif /* CONFIG_RTW_CFGVENDOR_LLSTATS */
 
-		RTW_PRINT("nolinked power save leave\n");
+		//RTW_PRINT("nolinked power save leave\n");
 
 		RTW_INFO("==> ips_leave.....LED(0x%08x)...\n", rtw_read32(padapter, 0x4c));
 		pwrpriv->bips_processing = _FALSE;
diff --git a/drivers/net/wireless/rockchip_wlan/rtl8821cs/core/rtw_recv.c b/drivers/net/wireless/rockchip_wlan/rtl8821cs/core/rtw_recv.c
index e68c5b30d..0b470e5fe 100644
--- a/drivers/net/wireless/rockchip_wlan/rtl8821cs/core/rtw_recv.c
+++ b/drivers/net/wireless/rockchip_wlan/rtl8821cs/core/rtw_recv.c
@@ -3634,13 +3634,8 @@ int validate_mp_recv_frame(_adapter *adapter, union recv_frame *precv_frame)
 		}
 
 		if (bDumpRxPkt == 1) { /* dump all rx packets */
-			int i;
 			RTW_INFO("############ type:0x%02x subtype:0x%02x #################\n", type, subtype);
 
-			for (i = 0; i < precv_frame->u.hdr.len; i = i + 8)
-				RTW_INFO("%02X:%02X:%02X:%02X:%02X:%02X:%02X:%02X:\n", *(ptr + i),
-					*(ptr + i + 1), *(ptr + i + 2) , *(ptr + i + 3) , *(ptr + i + 4), *(ptr + i + 5), *(ptr + i + 6), *(ptr + i + 7));
-				RTW_INFO("#############################\n");
 				_rtw_memset(pmppriv->mplink_buf, '\0' , sizeof(pmppriv->mplink_buf));
 				_rtw_memcpy(pmppriv->mplink_buf, ptr, precv_frame->u.hdr.len);
 				pmppriv->mplink_rx_len = precv_frame->u.hdr.len;
diff --git a/drivers/net/wireless/rockchip_wlan/rtl8821cs/core/rtw_wlan_util.c b/drivers/net/wireless/rockchip_wlan/rtl8821cs/core/rtw_wlan_util.c
index e162b0867..539a88f92 100644
--- a/drivers/net/wireless/rockchip_wlan/rtl8821cs/core/rtw_wlan_util.c
+++ b/drivers/net/wireless/rockchip_wlan/rtl8821cs/core/rtw_wlan_util.c
@@ -17,7 +17,7 @@
 
 #include <drv_types.h>
 #include <hal_data.h>
-
+MODULE_IMPORT_NS(VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver);
 #if defined(CONFIG_WOWLAN) || defined(CONFIG_AP_WOWLAN)
 	#include <linux/inetdevice.h>
 	#define ETH_TYPE_OFFSET	12
diff --git a/drivers/net/wireless/rockchip_wlan/rtl8821cs/hal/phydm/halrf/rtl8821c/halhwimg8821c_rf.c b/drivers/net/wireless/rockchip_wlan/rtl8821cs/hal/phydm/halrf/rtl8821c/halhwimg8821c_rf.c
index 8e1ffa7e3..1e9dbc879 100644
--- a/drivers/net/wireless/rockchip_wlan/rtl8821cs/hal/phydm/halrf/rtl8821c/halhwimg8821c_rf.c
+++ b/drivers/net/wireless/rockchip_wlan/rtl8821cs/hal/phydm/halrf/rtl8821c/halhwimg8821c_rf.c
@@ -29,15 +29,15 @@
 
 #define ODM_WIN 0x08
 
-#if (DM_ODM_SUPPORT_TYPE == ODM_WIN)
-#if RT_PLATFORM == PLATFORM_MACOSX
-#include "phydm_precomp.h"
-#else
-#include "../phydm_precomp.h"
-#endif
-#else
+//#if (DM_ODM_SUPPORT_TYPE == ODM_WIN)
+//#if RT_PLATFORM == PLATFORM_MACOSX
+//#include "phydm_precomp.h"
+//#else
+//#include "../phydm_precomp.h"
+//#endif
+//#else
 #include "../../phydm_precomp.h"
-#endif
+//#endif
 
 #define D_S_SIZE DELTA_SWINGIDX_SIZE
 #define D_ST_SIZE DELTA_SWINTSSI_SIZE
diff --git a/drivers/net/wireless/rockchip_wlan/rtl8821cs/hal/phydm/halrf/rtl8821c/halrf_8821c.c b/drivers/net/wireless/rockchip_wlan/rtl8821cs/hal/phydm/halrf/rtl8821c/halrf_8821c.c
index c08d095ad..0612523b7 100644
--- a/drivers/net/wireless/rockchip_wlan/rtl8821cs/hal/phydm/halrf/rtl8821c/halrf_8821c.c
+++ b/drivers/net/wireless/rockchip_wlan/rtl8821cs/hal/phydm/halrf/rtl8821c/halrf_8821c.c
@@ -15,15 +15,17 @@
  *****************************************************************************/
 
 #include "mp_precomp.h"
-#if (DM_ODM_SUPPORT_TYPE == ODM_WIN)
-#if RT_PLATFORM == PLATFORM_MACOSX
+//#if (DM_ODM_SUPPORT_TYPE == ODM_WIN)
+
+//#if RT_PLATFORM == PLATFORM_MACOSX
+//#include "phydm_precomp.h"
+//#else
+//#include "../phydm_precomp.h"
+//#endif
+
+//#else
 #include "phydm_precomp.h"
-#else
-#include "../phydm_precomp.h"
-#endif
-#else
-#include "../../phydm_precomp.h"
-#endif
+//#endif
 
 #if (RTL8821C_SUPPORT == 1)
 void halrf_rf_lna_setting_8821c(struct dm_struct *dm_void,
diff --git a/drivers/net/wireless/rockchip_wlan/rtl8821cs/hal/phydm/halrf/rtl8821c/halrf_iqk_8821c.c b/drivers/net/wireless/rockchip_wlan/rtl8821cs/hal/phydm/halrf/rtl8821c/halrf_iqk_8821c.c
index fa8adba63..ba66b241a 100644
--- a/drivers/net/wireless/rockchip_wlan/rtl8821cs/hal/phydm/halrf/rtl8821c/halrf_iqk_8821c.c
+++ b/drivers/net/wireless/rockchip_wlan/rtl8821cs/hal/phydm/halrf/rtl8821c/halrf_iqk_8821c.c
@@ -15,15 +15,15 @@
  *****************************************************************************/
 
 #include "mp_precomp.h"
-#if (DM_ODM_SUPPORT_TYPE == ODM_WIN)
-#if RT_PLATFORM == PLATFORM_MACOSX
+//#if (DM_ODM_SUPPORT_TYPE == ODM_WIN)
+//#if RT_PLATFORM == PLATFORM_MACOSX
+//#include "phydm_precomp.h"
+//#else
+//#include "../phydm_precomp.h"
+//#endif
+//#else
 #include "phydm_precomp.h"
-#else
-#include "../phydm_precomp.h"
-#endif
-#else
-#include "../../phydm_precomp.h"
-#endif
+//#endif
 
 #if (RTL8821C_SUPPORT == 1)
 /*---------------------------Define Local Constant---------------------------*/
diff --git a/drivers/net/wireless/rockchip_wlan/rtl8821cs/hal/phydm/phydm_adc_sampling.c b/drivers/net/wireless/rockchip_wlan/rtl8821cs/hal/phydm/phydm_adc_sampling.c
index 371cb9a2d..a7ba1a50f 100644
--- a/drivers/net/wireless/rockchip_wlan/rtl8821cs/hal/phydm/phydm_adc_sampling.c
+++ b/drivers/net/wireless/rockchip_wlan/rtl8821cs/hal/phydm/phydm_adc_sampling.c
@@ -1260,9 +1260,9 @@ void phydm_la_set_buff_mode(void *dm_void, enum la_buff_mode mode)
 	buf->start_pos = end_pos_tmp - buf->buffer_size;
 	smp->smp_number_max = buf->buffer_size >> 3;
 
-	pr_debug("start_addr=(0x%x), end_addr=(0x%x), buffer_size=(0x%x), smp_number_max=(%d)\n",
+	/*pr_debug("start_addr=(0x%x), end_addr=(0x%x), buffer_size=(0x%x), smp_number_max=(%d)\n",
 		 buf->start_pos, buf->end_pos, buf->buffer_size,
-		 smp->smp_number_max);
+		 smp->smp_number_max);*/
 }
 
 void phydm_la_adc_smp_start(void *dm_void)
diff --git a/drivers/net/wireless/rockchip_wlan/rtl8821cs/hal/rtl8821c/rtl8821c_ops.c b/drivers/net/wireless/rockchip_wlan/rtl8821cs/hal/rtl8821c/rtl8821c_ops.c
index 9e0b9eae1..c219cdce4 100644
--- a/drivers/net/wireless/rockchip_wlan/rtl8821cs/hal/rtl8821c/rtl8821c_ops.c
+++ b/drivers/net/wireless/rockchip_wlan/rtl8821cs/hal/rtl8821c/rtl8821c_ops.c
@@ -1891,6 +1891,7 @@ u8 rtl8821c_sethwreg(PADAPTER adapter, u8 variable, u8 *val)
 	case HW_VAR_H2C_INACTIVE_IPS:
 #ifdef CONFIG_WOWLAN
 		rtl8821c_set_fw_pwrmode_inips_cmd_wowlan(adapter, *val);
+		break;
 #endif /* CONFIG_WOWLAN */
 #ifdef CONFIG_WOWLAN
 	case HW_VAR_VENDOR_WOW_MODE:
diff --git a/drivers/net/wireless/rockchip_wlan/rtl8821cs/include/drv_conf.h b/drivers/net/wireless/rockchip_wlan/rtl8821cs/include/drv_conf.h
index 91393eb35..2a3f889df 100644
--- a/drivers/net/wireless/rockchip_wlan/rtl8821cs/include/drv_conf.h
+++ b/drivers/net/wireless/rockchip_wlan/rtl8821cs/include/drv_conf.h
@@ -166,13 +166,14 @@
 	#undef CONFIG_RESUME_IN_WORKQUEUE
 #endif
 */
-
+#if 0
 #ifdef CONFIG_RESUME_IN_WORKQUEUE /* this can be removed, because there is no case for this... */
 	#if !defined(CONFIG_WAKELOCK) && !defined(CONFIG_ANDROID_POWER)
 		#error "enable CONFIG_RESUME_IN_WORKQUEUE without CONFIG_WAKELOCK or CONFIG_ANDROID_POWER will suffer from the danger of wifi's unfunctionality..."
 		#error "If you still want to enable CONFIG_RESUME_IN_WORKQUEUE in this case, mask this preprossor checking and GOOD LUCK..."
 	#endif
 #endif
+#endif
 
 /* About USB VENDOR REQ */
 #if defined(CONFIG_USB_VENDOR_REQ_BUFFER_PREALLOC) && !defined(CONFIG_USB_VENDOR_REQ_MUTEX)
diff --git a/drivers/net/wireless/rockchip_wlan/rtl8821cs/os_dep/osdep_service.c b/drivers/net/wireless/rockchip_wlan/rtl8821cs/os_dep/osdep_service.c
index 70fb6d513..186ca889f 100644
--- a/drivers/net/wireless/rockchip_wlan/rtl8821cs/os_dep/osdep_service.c
+++ b/drivers/net/wireless/rockchip_wlan/rtl8821cs/os_dep/osdep_service.c
@@ -28,6 +28,7 @@ atomic_t _malloc_size = ATOMIC_INIT(0);
 #endif
 #endif /* DBG_MEMORY_LEAK */
 
+MODULE_IMPORT_NS(VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver);
 
 #if defined(PLATFORM_LINUX)
 /*
@@ -2482,7 +2483,7 @@ static int writeFile(struct file *fp, char *buf, int len)
 static int isDirReadable(const char *pathname, u32 *sz)
 {
 	struct path path;
-	int error = 0;
+	//int error = 0;
 
 	return kern_path(pathname, LOOKUP_FOLLOW, &path);
 }
