diff --git a/arch/arm64/boot/dts/rockchip/rk3568-android.dtsi b/arch/arm64/boot/dts/rockchip/rk3568-android.dtsi
index 87c4ba3f6..b71785476 100644
--- a/arch/arm64/boot/dts/rockchip/rk3568-android.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3568-android.dtsi
@@ -6,7 +6,7 @@
 
 / {
 	chosen: chosen {
-		bootargs = "earlycon=uart8250,mmio32,0xfe660000 console=ttyFIQ0";
+		bootargs = "earlycon=uart8250,mmio32,0xfe660000 console=ttyFIQ0 irqchip.gicv3_pseudo_nmi=0";
 	};
 
 	aliases {
diff --git a/arch/arm64/boot/dts/rockchip/rk3568-evb.dtsi b/arch/arm64/boot/dts/rockchip/rk3568-evb.dtsi
index f13c2387f..a9d89b2ea 100644
--- a/arch/arm64/boot/dts/rockchip/rk3568-evb.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3568-evb.dtsi
@@ -151,7 +151,7 @@
 		status = "disabled";
 		compatible = "simple-audio-card";
 		simple-audio-card,format = "dsp_a";
-		simple-audio-card,bitclock-inversion;
+		simple-audio-card,bitclock-inversion = <0>;
 		simple-audio-card,mclk-fs = <256>;
 		simple-audio-card,name = "rockchip,bt";
 		simple-audio-card,cpu {
@@ -189,7 +189,7 @@
 	leds: leds {
 		compatible = "gpio-leds";
 		work_led: work {
-			gpios = <&gpio0 RK_PC0 GPIO_ACTIVE_HIGH>;
+			gpios = <&gpio0 RK_PC4 GPIO_ACTIVE_LOW>;
 			linux,default-trigger = "heartbeat";
 		};
 	};
@@ -213,7 +213,7 @@
 	};
 
 	rk809_sound: rk809-sound {
-		status = "okay";
+		status = "disabled";
 		compatible = "rockchip,multicodecs-card";
 		rockchip,card-name = "rockchip-rk809";
 		hp-det-gpio = <&gpio3 RK_PA1 GPIO_ACTIVE_LOW>;
@@ -226,7 +226,7 @@
 	};
 
 	spdif-sound {
-		status = "okay";
+		status = "disabled";
 		compatible = "simple-audio-card";
 		simple-audio-card,name = "ROCKCHIP,SPDIF";
 		simple-audio-card,mclk-fs = <128>;
@@ -239,7 +239,7 @@
 	};
 
 	spdif_out: spdif-out {
-			status = "okay";
+			status = "disabled";
 			compatible = "linux,spdif-dit";
 			#sound-dai-cells = <0>;
 	};
@@ -294,6 +294,9 @@
 		vin-supply = <&vcc5v0_usb>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&vcc5v0_host_en>;
+		regulator-state-mem {
+			regulator-on-in-suspend;
+		};
 	};
 
 	vcc5v0_otg: vcc5v0-otg-regulator {
@@ -306,14 +309,17 @@
 		vin-supply = <&vcc5v0_usb>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&vcc5v0_otg_en>;
+		regulator-boot-on;
+		regulator-always-on;
 	};
 
 	vcc3v3_lcd0_n: vcc3v3-lcd0-n {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc3v3_lcd0_n";
 		regulator-boot-on;
+		regulator-always-on;
 		regulator-state-mem {
-			regulator-off-in-suspend;
+			regulator-on-in-suspend;
 		};
 	};
 
@@ -341,6 +347,7 @@
 		 */
 		post-power-on-delay-ms = <200>;
 		reset-gpios = <&gpio3 RK_PD5 GPIO_ACTIVE_LOW>;
+		status = "disabled";
 	};
 
 	wireless_wlan: wireless-wlan {
@@ -348,7 +355,7 @@
 		rockchip,grf = <&grf>;
 		wifi_chip_type = "ap6398s";
 		WIFI,poweren_gpio = <&gpio3 RK_PD5 GPIO_ACTIVE_HIGH>;
-		status = "okay";
+		status = "disabled";
 	};
 
 	wireless_bluetooth: wireless-bluetooth {
@@ -363,12 +370,21 @@
 		BT,reset_gpio    = <&gpio3 RK_PA0 GPIO_ACTIVE_HIGH>;
 		BT,wake_gpio     = <&gpio3 RK_PA1 GPIO_ACTIVE_HIGH>;
 		BT,wake_host_irq = <&gpio3 RK_PA2 GPIO_ACTIVE_HIGH>;
-		status = "okay";
+		status = "disabled";
 	};
 
 	test-power {
 		status = "okay";
 	};
+
+
+        gpio_group{
+                compatible = "gpio-group";
+		pinctrl-names = "gpio-watchdog";
+		watchdog-gpios = <&gpio4 RK_PB7 GPIO_ACTIVE_HIGH>;
+		empty-property-xpi3568;
+		status = "okay";
+	};
 };
 
 &bus_npu {
@@ -377,10 +393,38 @@
 	status = "okay";
 };
 
+&can0 {
+	assigned-clocks = <&cru CLK_CAN0>;
+	assigned-clock-rates = <150000000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&can0m1_pins>;
+	status = "disabled";
+};
+
+&can1 {
+	assigned-clocks = <&cru CLK_CAN1>;
+	assigned-clock-rates = <150000000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&can1m1_pins>;
+	status = "disabled";
+};
+
+&can2 {
+	assigned-clocks = <&cru CLK_CAN2>;
+	assigned-clock-rates = <150000000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&can2m1_pins>;
+	status = "disabled";
+};
+
 &cpu0 {
 	cpu-supply = <&vdd_cpu>;
 };
 
+&thermal_zones {
+	status = "okay";
+};
+
 &dfi {
 	status = "okay";
 };
@@ -397,18 +441,16 @@
 		status = "okay";
 		compatible = "simple-panel-dsi";
 		reg = <0>;
-		backlight = <&backlight>;
+		//backlight = <&backlight>;
 		reset-delay-ms = <60>;
 		enable-delay-ms = <60>;
 		prepare-delay-ms = <60>;
 		unprepare-delay-ms = <60>;
 		disable-delay-ms = <60>;
-		width-mm = <68>;
-		height-mm = <121>;
 		dsi,flags = <(MIPI_DSI_MODE_VIDEO | MIPI_DSI_MODE_VIDEO_BURST |
 			MIPI_DSI_MODE_LPM | MIPI_DSI_MODE_EOT_PACKET)>;
 		dsi,format = <MIPI_DSI_FMT_RGB888>;
-		dsi,lanes  = <4>;
+		dsi,lanes  = <2>;
 		panel-init-sequence = [
 			23 00 02 FE 21
 			23 00 02 04 00
@@ -681,15 +723,15 @@
 		disp_timings0: display-timings {
 			native-mode = <&dsi0_timing0>;
 			dsi0_timing0: timing0 {
-				clock-frequency = <132000000>;
+				clock-frequency = <27448000>;
 				hactive = <1080>;
 				vactive = <1920>;
-				hfront-porch = <15>;
-				hsync-len = <2>;
-				hback-porch = <30>;
-				vfront-porch = <15>;
+				hfront-porch = <70>;
+				hsync-len = <20>;
+				hback-porch = <26>;
+				vfront-porch = <7>;
 				vsync-len = <2>;
-				vback-porch = <15>;
+				vback-porch = <21>;
 				hsync-active = <0>;
 				vsync-active = <0>;
 				de-active = <0>;
@@ -728,7 +770,7 @@
 	status = "disabled";
 	//rockchip,lane-rate = <1000>;
 	dsi1_panel: panel@0 {
-		status = "okay";
+		status = "disabled";
 		compatible = "simple-panel-dsi";
 		reg = <0>;
 		backlight = <&backlight1>;
@@ -737,8 +779,6 @@
 		prepare-delay-ms = <60>;
 		unprepare-delay-ms = <60>;
 		disable-delay-ms = <60>;
-		width-mm = <68>;
-		height-mm = <121>;
 		dsi,flags = <(MIPI_DSI_MODE_VIDEO | MIPI_DSI_MODE_VIDEO_BURST |
 			MIPI_DSI_MODE_LPM | MIPI_DSI_MODE_EOT_PACKET)>;
 		dsi,format = <MIPI_DSI_FMT_RGB888>;
@@ -1088,6 +1128,21 @@
 
 &i2c0 {
 	status = "okay";
+	rtc-hym8563s@51 {
+		compatible = "haoyu,hym8563";
+		reg = <0x51>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&rtc_int>;
+		gpio_intb = <&gpio4 RK_PC0 GPIO_ACTIVE_LOW>;
+		init_date = "2015/01/01";
+		status = "okay";
+	};
+
+	eeprom@50 {
+		compatible = "atmel,24c02";
+		reg = <0x50>;
+		status="okay";
+	};
 
 	vdd_cpu: tcs4525@1c {
 		compatible = "tcs,tcs4525";
@@ -1180,7 +1235,7 @@
 				regulator-initial-mode = <0x2>;
 				regulator-name = "vdd_logic";
 				regulator-state-mem {
-					regulator-off-in-suspend;
+					regulator-on-in-suspend;
 				};
 			};
 
@@ -1240,7 +1295,8 @@
 				regulator-max-microvolt = <900000>;
 				regulator-name = "vdda_0v9";
 				regulator-state-mem {
-					regulator-off-in-suspend;
+					regulator-on-in-suspend;
+					regulator-suspend-microvolt = <900000>;
 				};
 			};
 
@@ -1297,7 +1353,8 @@
 				regulator-max-microvolt = <1800000>;
 				regulator-name = "vcca_1v8";
 				regulator-state-mem {
-					regulator-off-in-suspend;
+					regulator-on-in-suspend;
+					regulator-suspend-microvolt = <1800000>;
 				};
 			};
 
@@ -1320,7 +1377,8 @@
 				regulator-max-microvolt = <1800000>;
 				regulator-name = "vcca1v8_image";
 				regulator-state-mem {
-					regulator-off-in-suspend;
+					regulator-on-in-suspend;
+					regulator-suspend-microvolt = <1800000>;
 				};
 			};
 
@@ -1340,7 +1398,7 @@
 				regulator-boot-on;
 				regulator-name = "vcc_3v3";
 				regulator-state-mem {
-					regulator-off-in-suspend;
+					regulator-on-in-suspend;
 				};
 			};
 
@@ -1367,29 +1425,16 @@
 			hp-volume = <20>;
 			spk-volume = <3>;
 			mic-in-differential;
-			status = "okay";
+			status = "disabled";
 		};
 	};
 };
 
-&i2c1 {
-	status = "okay";
-
-	gt1x: gt1x@14 {
-		compatible = "goodix,gt1x";
-		reg = <0x14>;
-		pinctrl-names = "default";
-		pinctrl-0 = <&touch_gpio>;
-		goodix,rst-gpio = <&gpio0 RK_PB6 GPIO_ACTIVE_HIGH>;
-		goodix,irq-gpio = <&gpio0 RK_PB5 IRQ_TYPE_LEVEL_LOW>;
-	};
-};
-
 &i2c5 {
-	status = "okay";
+	status = "disabled";
 
 	mxc6655xa: mxc6655xa@15 {
-		status = "okay";
+		status = "disabled";
 		compatible = "gs_mxc6655xa";
 		pinctrl-names = "default";
 		pinctrl-0 = <&mxc6655xa_irq_gpio>;
@@ -1408,7 +1453,7 @@
 };
 
 &i2s1_8ch {
-	status = "okay";
+	status = "disabled";
 	rockchip,clk-trcm = <1>;
 	pinctrl-names = "default";
 	pinctrl-0 = <&i2s1m0_sclktx
@@ -1522,6 +1567,12 @@
 			rockchip,pins = <2 RK_PB1 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
 	};
+
+	rtc {
+		rtc_int: rtc-int {
+			rockchip,pins = <4 RK_PC0 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
 };
 
  /*
@@ -1542,20 +1593,25 @@
   */
 &pmu_io_domains {
 	status = "okay";
+	pmuio1-supply = <&vcc3v3_pmu>;
 	pmuio2-supply = <&vcc3v3_pmu>;
-	vccio1-supply = <&vccio_acodec>;
+	vccio1-supply = <&vcc_3v3>;
 	vccio3-supply = <&vccio_sd>;
-	vccio4-supply = <&vcc_3v3>;
+	vccio4-supply = <&vcc_1v8>;
 	vccio5-supply = <&vcc_3v3>;
 	vccio6-supply = <&vcc_3v3>;
 	vccio7-supply = <&vcc_3v3>;
 };
 
 &pwm4 {
-	status = "okay";
+	status = "disabled";
 };
 
 &pwm5 {
+	status = "disabled";
+};
+
+&pwm6 {
 	status = "okay";
 };
 
@@ -1563,7 +1619,7 @@
 	status = "okay";
 
 	compatible = "rockchip,remotectl-pwm";
-	remote_pwm_id = <3>;
+	remote_pwm_id = <7>;
 	handle_cpu_id = <1>;
 	remote_support_psci = <0>;
 	pinctrl-names = "default";
@@ -1662,6 +1718,62 @@
 			<0xb6	KEY_0>,
 			<0xb5	KEY_BACKSPACE>;
 	};
+
+	ir_key4 {
+                rockchip,usercode = <0xbd02>;
+                rockchip,key_table =
+             <0x2f   KEY_BACK>,
+             <0x17   KEY_BACK>,
+             <0xb    KEY_BACK>,
+             <0x35   KEY_UP>,
+             <0x31   KEY_ENTER>,
+             <0x2d   KEY_DOWN>,
+             <0x66   KEY_LEFT>,
+             <0x3e   KEY_RIGHT>,
+             <0x6a   KEY_HOME>,
+             <0x22   KEY_VOLUMEUP>,
+             <0x73   KEY_VOLUMEDOWN>,
+             <0x39   KEY_VOLUMEDOWN>,
+             <0xba   KEY_DISPLAY_OFF>,
+             <0x7f   KEY_MUTE>,
+             <0x63   KEY_MENU>,
+             <0x31   KEY_REPLY>,
+             <0xa8   KEY_SEARCH>,
+             <0xaf   KEY_TEXT>,
+             <0xae   KEY_APPSELECT>,
+             <0x3a   KEY_INFO>,
+             <0x65 KEY_SCROLLUP>,
+             <0x32 KEY_SCROLLDOWN>,
+             <0xb9 KEY_RED>,
+             <0xb8 KEY_GREEN>,
+             <0xb7 KEY_YELLOW>,
+             <0xb6 KEY_BLUE>,
+             <0x27 KEY_WWW>,
+             <0xa3 KEY_EDIT>,
+             <0xa6 KEY_FN_F1>,
+             <0xa5 KEY_FN_F2>,
+             <0xa4 KEY_FN_F3>,
+             <0xa9 BTN_MOUSE>,
+             <0xa8 KEY_FN_F4>,
+             <0xad KEY_REWIND>,
+             <0xbc KEY_PLAYPAUSE>,
+             <0xac KEY_FASTFORWARD>,
+             <0xab KEY_STOPCD>,
+             <0x29 KEY_ZOOMIN>,
+             <0x76 KEY_ZOOMOUT>,
+             <0x6d KEY_1>,
+             <0x6c KEY_2>,
+             <0x33 KEY_3>,
+             <0x71 KEY_4>,
+             <0x70 KEY_5>,
+             <0x37 KEY_6>,
+             <0x75 KEY_7>,
+             <0x74 KEY_8>,
+             <0x3b KEY_9>,
+             <0x78 KEY_0>,
+             <0x26 KEY_CHAT>,
+             <0x7d KEY_DOT>;
+        };
 };
 
 &rk_rga {
@@ -1714,10 +1826,14 @@
 	status = "okay";
 };
 
+&rockchip_suspend {
+	status = "okay";
+};
+
 &sdmmc0 {
 	max-frequency = <150000000>;
-	no-sdio;
-	no-mmc;
+	supports-sd;
+	supports-emmc;
 	bus-width = <4>;
 	cap-mmc-highspeed;
 	cap-sd-highspeed;
@@ -1743,7 +1859,7 @@
 };
 
 &spdif_8ch {
-	status = "okay";
+	status = "disabled";
 };
 
 &tsadc {
@@ -1837,8 +1953,8 @@
 
 &vop {
 	status = "okay";
-	assigned-clocks = <&cru DCLK_VOP0>, <&cru DCLK_VOP1>, <&cru DCLK_VOP2>;
-	assigned-clock-parents = <&pmucru PLL_HPLL>, <&cru PLL_VPLL>, <&cru PLL_GPLL>;
+	assigned-clocks = <&cru DCLK_VOP0>, <&cru DCLK_VOP1>;
+	assigned-clock-parents = <&pmucru PLL_HPLL>, <&cru PLL_VPLL>;
 };
 
 &vop_mmu {
diff --git a/arch/arm64/boot/dts/rockchip/rk3568-evb1-ddr4-v10.dtsi b/arch/arm64/boot/dts/rockchip/rk3568-evb1-ddr4-v10.dtsi
index dc9ad4236..837dff63f 100644
--- a/arch/arm64/boot/dts/rockchip/rk3568-evb1-ddr4-v10.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3568-evb1-ddr4-v10.dtsi
@@ -61,7 +61,7 @@
 		regulator-min-microvolt = <3300000>;
 		regulator-max-microvolt = <3300000>;
 		enable-active-high;
-		gpio = <&gpio0 RK_PD4 GPIO_ACTIVE_HIGH>;
+//		gpio = <&gpio0 RK_PD4 GPIO_ACTIVE_HIGH>;
 		startup-delay-us = <5000>;
 		vin-supply = <&dc_12v>;
 	};
@@ -76,7 +76,7 @@
 		vin-supply = <&vcc5v0_sys>;
 	};
 
-	vcc_camera: vcc-camera-regulator {
+	/*vcc_camera: vcc-camera-regulator {
 		compatible = "regulator-fixed";
 		gpio = <&gpio0 RK_PC1 GPIO_ACTIVE_HIGH>;
 		pinctrl-names = "default";
@@ -85,6 +85,13 @@
 		enable-active-high;
 		regulator-always-on;
 		regulator-boot-on;
+	};*/
+
+	ext_cam_clk: external-camera-clock {
+		compatible = "fixed-clock";
+		clock-frequency = <25000000>;
+		clock-output-names = "CLK_CAMERA_25MHZ";
+		#clock-cells = <0>;
 	};
 };
 
@@ -115,19 +122,9 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 
-			mipi_in_ucam0: endpoint@1 {
+			dphy0_in: endpoint@1 {
 				reg = <1>;
-				remote-endpoint = <&ucam_out0>;
-				data-lanes = <1 2 3 4>;
-			};
-			mipi_in_ucam1: endpoint@2 {
-				reg = <2>;
-				remote-endpoint = <&gc8034_out>;
-				data-lanes = <1 2 3 4>;
-			};
-			mipi_in_ucam2: endpoint@3 {
-				reg = <3>;
-				remote-endpoint = <&ov5695_out>;
+				remote-endpoint = <&ov5647_out>;
 				data-lanes = <1 2>;
 			};
 		};
@@ -136,7 +133,7 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 
-			csidphy_out: endpoint@0 {
+			csidphy0_out: endpoint@0 {
 				reg = <0>;
 				remote-endpoint = <&isp0_in>;
 			};
@@ -181,20 +178,21 @@
 };
 
 &dsi1_panel {
+	status = "disabled";
 	power-supply = <&vcc3v3_lcd1_n>;
 };
 
 &edp {
 	hpd-gpios = <&gpio0 RK_PC2 GPIO_ACTIVE_HIGH>;
-	status = "okay";
+	status = "disabled";
 };
 
 &edp_phy {
-	status = "okay";
+	status = "disabled";
 };
 
 &edp_in_vp0 {
-	status = "okay";
+	status = "disabled";
 };
 
 &edp_in_vp1 {
@@ -214,6 +212,8 @@
 	assigned-clock-parents = <&cru SCLK_GMAC0_RGMII_SPEED>, <&cru CLK_MAC0_2TOP>;
 	assigned-clock-rates = <0>, <125000000>;
 
+	empty-property-xpi;
+
 	pinctrl-names = "default";
 	pinctrl-0 = <&gmac0_miim
 		     &gmac0_tx_bus2
@@ -252,19 +252,55 @@
 	rx_delay = <0x26>;
 
 	phy-handle = <&rgmii_phy1>;
-	status = "okay";
+	status = "disabled";
 };
 
-/*
- * power-supply should switche to vcc3v3_lcd1_n
- * when mipi panel is connected to dsi1.
- */
-&gt1x {
-	power-supply = <&vcc3v3_lcd0_n>;
-};
+ 
+&i2c1 {
+        status = "okay";
+        pinctrl-names = "default";
+        pinctrl-0 = <&i2c1_xfer>;
+        clock-frequency = <100000>;
+	camera0: ov5647@36 {
+                compatible = "ovti,ov5647";
+                reg = <0x36>;
+                clocks = <&ext_cam_clk>;
+                pwdn-gpios = <&gpio2 RK_PD0 GPIO_ACTIVE_LOW>;
+                rockchip,camera-module-index = <0>;
+                rockchip,camera-module-facing = "back";
+                rockchip,camera-module-name = "TongJu";
+                rockchip,camera-module-lens-name = "CHT842-MD";
+                status = "okay";
+                port {
+                        ov5647_out: endpoint {
+                                remote-endpoint = <&dphy0_in>;
+                                data-lanes = <1 2>;
+                        };
+                };
+        };
+};
+
+ &i2c2 {
+         status = "okay";
+         pinctrl-names = "default";
+	pinctrl-0 = <&i2c2m0_xfer>;
+	clock-frequency = <100000>;
+ 
+         touchscreen: tinker_ft5406@38 {
+                       compatible = "tinker_ft5406";
+                       reg = <0x38>;
+                       status = "okay";
+               };
+               tinker_mcu: tinker_mcu@45 {
+                       compatible = "tinker_mcu";
+                       reg = <0x45>;
+                       status = "okay";
+               };
+ };
 
 &i2c4 {
-	status = "okay";
+	status = "disabled";
+	#if 0
 	gc8034: gc8034@37 {
 		compatible = "galaxycore,gc8034";
 		status = "okay";
@@ -273,7 +309,6 @@
 		clock-names = "xvclk";
 		pinctrl-names = "default";
 		pinctrl-0 = <&cif_clk>;
-		power-domains = <&power RK3568_PD_VI>;
 		reset-gpios = <&gpio3 RK_PB6 GPIO_ACTIVE_LOW>;
 		pwdn-gpios = <&gpio4 RK_PB4 GPIO_ACTIVE_LOW>;
 		rockchip,grf = <&grf>;
@@ -331,6 +366,8 @@
 			};
 		};
 	};
+	#endif
+	
 };
 
 &mdio0 {
@@ -356,13 +393,13 @@
 };
 
 &pcie30phy {
-	status = "okay";
+	status = "disabled";
 };
 
 &pcie3x2 {
 	reset-gpios = <&gpio2 RK_PD6 GPIO_ACTIVE_HIGH>;
 	vpcie3v3-supply = <&vcc3v3_pcie>;
-	status = "okay";
+	status = "disabled";
 };
 
 &pinctrl {
@@ -392,10 +429,6 @@
 	};
 };
 
-&rk809_sound {
-	hp-det-gpio = <&gpio3 RK_PC2 GPIO_ACTIVE_LOW>;
-};
-
 &rkisp {
 	status = "okay";
 };
@@ -413,7 +446,7 @@
 
 		isp0_in: endpoint@0 {
 			reg = <0>;
-			remote-endpoint = <&csidphy_out>;
+			remote-endpoint = <&csidphy0_out>;
 		};
 	};
 };
@@ -424,12 +457,12 @@
 };
 
 &route_edp {
-	status = "okay";
+	status = "disabled";
 	connect = <&vp0_out_edp>;
 };
 
 &sata2 {
-	status = "okay";
+	status = "disabled";
 };
 
 &sdmmc2 {
@@ -446,28 +479,34 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&sdmmc2m0_bus4 &sdmmc2m0_cmd &sdmmc2m0_clk>;
 	sd-uhs-sdr104;
-	status = "okay";
+	status = "disabled";
 };
 
 &spdif_8ch {
-	status = "okay";
+	status = "disabled";
 	pinctrl-names = "default";
 	pinctrl-0 = <&spdifm1_tx>;
 };
 
 &uart8 {
-	status = "okay";
+	status = "disabled";
 	pinctrl-names = "default";
 	pinctrl-0 = <&uart8m0_xfer &uart8m0_ctsn>;
 };
 
+&uart3{
+	status = "okay";
+	pinctrl-names = "default";
+        pinctrl-0 = <&uart3m1_xfer>;
+};
+
 &vcc3v3_lcd0_n {
-	gpio = <&gpio0 RK_PC7 GPIO_ACTIVE_HIGH>;
-	enable-active-high;
+	gpio = <&gpio3 RK_PB5 GPIO_ACTIVE_LOW>;
+	//enable-active-high;
 };
 
 &vcc3v3_lcd1_n {
-	gpio = <&gpio0 RK_PC5 GPIO_ACTIVE_HIGH>;
+	//gpio = <&gpio0 RK_PC5 GPIO_ACTIVE_HIGH>;
 	enable-active-high;
 };
 
@@ -475,6 +514,7 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&wifi_host_wake_irq>;
 	WIFI,host_wake_irq = <&gpio3 RK_PD4 GPIO_ACTIVE_HIGH>;
+	status = "disabled";
 };
 
 &wireless_bluetooth {
@@ -489,5 +529,5 @@
 	BT,reset_gpio    = <&gpio3 RK_PA0 GPIO_ACTIVE_HIGH>;
 	BT,wake_gpio     = <&gpio3 RK_PA2 GPIO_ACTIVE_HIGH>;
 	BT,wake_host_irq = <&gpio3 RK_PA1 GPIO_ACTIVE_HIGH>;
-	status = "okay";
+	status = "disabled";
 };
diff --git a/arch/arm64/boot/dts/rockchip/rk3568-linux.dtsi b/arch/arm64/boot/dts/rockchip/rk3568-linux.dtsi
index c7e309645..47409fda9 100644
--- a/arch/arm64/boot/dts/rockchip/rk3568-linux.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3568-linux.dtsi
@@ -13,7 +13,7 @@
 	};
 
 	chosen: chosen {
-		bootargs = "earlycon=uart8250,mmio32,0xfe660000 console=ttyFIQ0 root=PARTUUID=614e0000-0000 rw rootwait";
+		bootargs = "earlycon=uart8250,mmio32,0xfe660000 console=ttyFIQ0";
 	};
 
 	fiq-debugger {
diff --git a/arch/arm64/boot/dts/rockchip/rk3568.dtsi b/arch/arm64/boot/dts/rockchip/rk3568.dtsi
index 778752440..ded42f205 100644
--- a/arch/arm64/boot/dts/rockchip/rk3568.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3568.dtsi
@@ -389,22 +389,22 @@
 
 	rockchip_suspend: rockchip-suspend {
 		compatible = "rockchip,pm-rk3568";
-		status = "disabled";
+		status = "okay";
 		rockchip,sleep-debug-en = <1>;
 		rockchip,sleep-mode-config = <
 			(0
-			| RKPM_SLP_ARMOFF_LOGOFF
 			| RKPM_SLP_CENTER_OFF
 			| RKPM_SLP_HW_PLLS_OFF
 			| RKPM_SLP_PMUALIVE_32K
-			| RKPM_SLP_OSC_DIS
-			| RKPM_SLP_PMIC_LP
 			| RKPM_SLP_32K_PVTM
 			)
 		>;
 		rockchip,wakeup-config = <
 			(0
 			| RKPM_GPIO_WKUP_EN
+			| RKPM_PWM0_WKUP_EN
+			| RKPM_USB_WKUP_EN
+			| RKPM_USB_LINESTATE_WKUP_EN
 			)
 		>;
 	};
diff --git a/arch/arm64/configs/rockchip_linux_defconfig b/arch/arm64/configs/rockchip_linux_defconfig
index 688203de6..4c2bddb59 100644
--- a/arch/arm64/configs/rockchip_linux_defconfig
+++ b/arch/arm64/configs/rockchip_linux_defconfig
@@ -598,6 +598,8 @@ CONFIG_VFAT_FS=y
 CONFIG_FAT_DEFAULT_CODEPAGE=936
 CONFIG_FAT_DEFAULT_IOCHARSET="utf8"
 CONFIG_NTFS_FS=y
+CONFIG_NTFS_DEBUG=y
+CONFIG_NTFS_RW=y
 CONFIG_TMPFS=y
 CONFIG_TMPFS_POSIX_ACL=y
 CONFIG_EFIVAR_FS=y
@@ -648,3 +650,69 @@ CONFIG_RCU_CPU_STALL_TIMEOUT=60
 CONFIG_FUNCTION_TRACER=y
 CONFIG_BLK_DEV_IO_TRACE=y
 CONFIG_LKDTM=y
+
+CONFIG_MEDIA_SUBDRV_AUTOSELECT=y
+CONFIG_MRVL8897U=y
+CONFIG_MRVL8897U_BT=y
+CONFIG_GPIO_CONTROL=y
+
+#PPP
+CONFIG_PPP=y
+CONFIG_PPP_BSDCOMP=y
+CONFIG_PPP_DEFLATE=y
+CONFIG_PPP_FILTER=y
+CONFIG_PPP_MPPE=y
+CONFIG_PPP_MULTILINK=y
+CONFIG_PPPOE=y
+CONFIG_PPP_ASYNC=y
+CONFIG_PPP_SYNC_TTY=y
+
+CONFIG_RTC_DRV_AM1805=y
+CONFIG_SND_SOC_ES8388=y
+CONFIG_EEPROM_AT24=y
+CONFIG_CAN=y
+CONFIG_CAN_ROCKCHIP=y
+CONFIG_CANFD_ROCKCHIP=y
+CONFIG_EXFAT_FS=y
+CONFIG_PM_SLEEP=y
+#wifi hotspot
+CONFIG_NETFILTER=y
+CONFIG_NF_CONNTRACK=y
+CONFIG_NF_TABLES=y
+CONFIG_NF_TABLES_INET=y
+CONFIG_NF_CONNTRACK_IPV4=y
+CONFIG_IP_NF_IPTABLES=y
+CONFIG_IP_NF_NAT=y
+CONFIG_IP_NF_TARGET_MASQUERADE=y
+
+#docker
+CONFIG_MEMCG=y
+CONFIG_VETH=y
+CONFIG_BRIDGE_NETFILTER=y
+CONFIG_NETFILTER_XT_MATCH_ADDRTYPE=y
+CONFIG_NETFILTER_XT_MATCH_CONNTRACK=y
+CONFIG_NETFILTER_XT_MATCH_IPVS=y
+CONFIG_NETFILTER_XT_MARK=y
+CONFIG_POSIX_MQUEUE=y
+CONFIG_CGROUP_BPF=y
+CONFIG_IP_NF_FILTER=y
+CONFIG_BLK_CGROUP=y
+#CONFIG_RT_GROUP_SCHED=y
+CONFIG_CGROUP_PIDS=y
+CONFIG_CGROUP_PERF=y
+CONFIG_BPF_SYSCALL=y
+CONFIG_BLK_DEV_THROTTLING=y
+CONFIG_INET_ESP=y
+CONFIG_NF_CONNTRACK_FTP=y
+CONFIG_NF_CONNTRACK_TFTP=y
+CONFIG_NF_TABLES_NETDEV=y
+CONFIG_IP_VS=y
+CONFIG_IP_VS_PROTO_TCP=y
+CONFIG_IP_VS_PROTO_UDP=y
+CONFIG_IP_VS_RR=y
+CONFIG_IP_VS_NFCT=y
+CONFIG_NF_TABLES_ARP=y
+
+CONFIG_TOUCHSCREEN_TINKER_FT5406=y
+CONFIG_TINKER_MCU=y
+CONFIG_VIDEO_OV5647=y
